core/src/main/scala/kafka/log/FileMessageSet.scala

```diff
@@ -260,20 +261,20 @@ class FileMessageSet private[kafka](@volatile var file: File,
         sizeOffsetBuffer.rewind()
         val offset = sizeOffsetBuffer.getLong()
         val size = sizeOffsetBuffer.getInt()
-        if(size < Message.MinMessageOverhead)
+        if(size < Message.MinMessageOverhead || location + SizeOffsetLength + size > end)
```

Comment:
Just to double-check: it is intentional to use `>` here even though we used `>=` a few lines above?